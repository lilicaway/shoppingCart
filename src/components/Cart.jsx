import React, { Component } from 'react';
import CartContent from './CartContent';
import CartAddProduct from './CartAddProduct';

class Cart extends Component {
  render() {
    return (
      <div>
        <CartContent />
        <CartAddProduct />
      </div>
    );
  }
}

export default Cart;
