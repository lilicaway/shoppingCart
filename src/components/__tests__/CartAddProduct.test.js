import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, render } from 'enzyme';
import { Provider } from 'react-redux';
import { Button } from 'react-bootstrap';
import { MemoryRouter } from 'react-router-dom';
import Cart, { UnwrappedCartAddProductTest } from '../CartAddProduct';

test('Cart add product renders correctly', () => {
  const products = [
    {
      id: '42d1595f-63d6-4c68-967a-e1ec4d7ca37c',
      name: 'Apple',
      price: 0.25
    },
    {
      id: 'a9dd5019-194f-4cd0-a661-5e9485676879',
      name: 'Papaya',
      price: 0.5,
      promotionName: '3 x 2'
    }
  ];
  const component = shallow(<UnwrappedCartAddProductTest products={products} />);
  expect(component).toMatchSnapshot();
});
