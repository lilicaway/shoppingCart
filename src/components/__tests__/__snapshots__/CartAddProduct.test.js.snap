// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart add product renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Panel
    bsClass="panel"
    bsStyle="default"
    defaultExpanded={false}
    header="Add Products to Cart"
>
    <Grid
        bsClass="container"
        componentClass="div"
        fluid={true}
    >
        <Connect(Product)
            id="42d1595f-63d6-4c68-967a-e1ec4d7ca37c"
            name="Apple"
            price={0.25}
        />
        <Connect(Product)
            id="a9dd5019-194f-4cd0-a661-5e9485676879"
            name="Papaya"
            price={0.5}
            promotionName="3 x 2"
        />
    </Grid>
</Panel>,
  "nodes": Array [
    <Panel
      bsClass="panel"
      bsStyle="default"
      defaultExpanded={false}
      header="Add Products to Cart"
>
      <Grid
            bsClass="container"
            componentClass="div"
            fluid={true}
      >
            <Connect(Product)
                  id="42d1595f-63d6-4c68-967a-e1ec4d7ca37c"
                  name="Apple"
                  price={0.25}
            />
            <Connect(Product)
                  id="a9dd5019-194f-4cd0-a661-5e9485676879"
                  name="Papaya"
                  price={0.5}
                  promotionName="3 x 2"
            />
      </Grid>
</Panel>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <CartAddProduct
        products={
                Array [
                        Object {
                          "id": "42d1595f-63d6-4c68-967a-e1ec4d7ca37c",
                          "name": "Apple",
                          "price": 0.25,
                        },
                        Object {
                          "id": "a9dd5019-194f-4cd0-a661-5e9485676879",
                          "name": "Papaya",
                          "price": 0.5,
                          "promotionName": "3 x 2",
                        },
                      ]
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": CartAddProduct {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "products": Array [
            Object {
              "id": "42d1595f-63d6-4c68-967a-e1ec4d7ca37c",
              "name": "Apple",
              "price": 0.25,
            },
            Object {
              "id": "a9dd5019-194f-4cd0-a661-5e9485676879",
              "name": "Papaya",
              "price": 0.5,
              "promotionName": "3 x 2",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Panel
          bsClass="panel"
          bsStyle="default"
          defaultExpanded={false}
          header="Add Products to Cart"
>
          <Grid
                    bsClass="container"
                    componentClass="div"
                    fluid={true}
          >
                    <Connect(Product)
                              id="42d1595f-63d6-4c68-967a-e1ec4d7ca37c"
                              name="Apple"
                              price={0.25}
                    />
                    <Connect(Product)
                              id="a9dd5019-194f-4cd0-a661-5e9485676879"
                              name="Papaya"
                              price={0.5}
                              promotionName="3 x 2"
                    />
          </Grid>
</Panel>,
        "_debugID": 2,
        "_renderedOutput": <Panel
          bsClass="panel"
          bsStyle="default"
          defaultExpanded={false}
          header="Add Products to Cart"
>
          <Grid
                    bsClass="container"
                    componentClass="div"
                    fluid={true}
          >
                    <Connect(Product)
                              id="42d1595f-63d6-4c68-967a-e1ec4d7ca37c"
                              name="Apple"
                              price={0.25}
                    />
                    <Connect(Product)
                              id="a9dd5019-194f-4cd0-a661-5e9485676879"
                              name="Papaya"
                              price={0.5}
                              promotionName="3 x 2"
                    />
          </Grid>
</Panel>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CartAddProduct
    products={
        Array [
            Object {
              "id": "42d1595f-63d6-4c68-967a-e1ec4d7ca37c",
              "name": "Apple",
              "price": 0.25,
            },
            Object {
              "id": "a9dd5019-194f-4cd0-a661-5e9485676879",
              "name": "Papaya",
              "price": 0.5,
              "promotionName": "3 x 2",
            },
          ]
    }
/>,
}
`;
